### PR TITLE
Enable other Frameworks

### PR DESCRIPTION
**Pull request by deinok**
This change should enable other frameworks to work

### CODE REVIEW

**Comment by jmarolf** [comment]
@deinok what is the motivation for this change? Global tools should be forward compatible with newer .NET Core SDK

**Comment by deinok** [comment]
In a `netcoreapp3.0` i can't add `dotnet-format` as a `<DotNetCliToolReference />`

**Comment by jmarolf** [comment]
@deinok ah, I see.  In .NET Core 3 `DotNetCliToolReference` is being deprecated  for local tools (early docs [here](https://github.com/dotnet/cli/issues/10288)). in .NET Core 3 you will just need to specify installation of .NET format in a `dotnet-tools.json` file and `dotnet restore` will work automatically

File: src/dotnet-format.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFrameworks>netcoreapp2.1;netcoreapp2.2;netcoreapp3.0</TargetFrameworks>
```
Comment (jmarolf) [inline src/dotnet-format.csproj]: we already specify that this tool can run on newer runtimes [here](https://github.com/dotnet/format/blob/master/src/runtimeconfig.template.json).  nuget not restoring it correcty in the 3.0 SDK is a bug that I am told is being fixed with the Local Tools SDK feature

Comment (wli3) [reply src/dotnet-format.csproj]: > nuget not restoring it correcty in the 3.0 SDK is a bug

It should not be able to directly restored by nuget (as in adding to csproj and run restore). It is designed that way. You could use "tool restore" with local tools.

**Comment by sharwell** [comment]
@deinok See https://github.com/dotnet/roslyn/pull/38562 for an example of this 😄

**Pull request closed without merging**